File: app/src/main/java/com/mss/imagesearcher/domain/repositories/ImagesNetRepository.java
package com.mss.imagesearcher.domain.repositories;

import androidx.annotation.NonNull;

import com.mss.imagesearcher.domain.models.ImageInfo;

import java.util.List;

import io.reactivex.Maybe;

public interface ImagesNetRepository {

    @NonNull
    Maybe<List<ImageInfo>> findImages(@NonNull final String query, int pageNumber);
}

File: app/src/main/java/com/mss/imagesearcher/model/repositories/ImagesNetRepository.kt
package com.mss.imagesearcher.model.repositories

import com.mss.imagesearcher.model.entity.ImageInfo
import com.mss.imagesearcher.model.entity.QueryParams

import io.reactivex.Maybe

interface ImagesNetRepository {

    fun findImages(query: QueryParams, pageNumber: Int): Maybe<List<ImageInfo>>
}
